docs/generate_imagelist.jl

```diff
@@ -52,28 +52,58 @@ function generate_imagelist(root)
         end
     end
 
+    # Refer to the metadata
+    path_toml = TestImages.image_path("metadata.toml")
+    metadata = Dict{String,String}.(TOML.parsefile(path_toml)["images"])
+
+    fullnames = ones(String, N)
+    notes = [String[] for i in 1:N]
+
+    for i in 1:N
+        name = splitext(filenames[i])[1]
+        j = findfirst(data -> data["name"]==name, metadata)
+        if !isnothing(j)
+            data = metadata[j]
+            if "fullname" in keys(data)
+                fullnames[i] = data["fullname"]
+            end
+            if "url" in keys(data)
+                url = data["url"]
+                push!(notes[i], "[origin]($(url))")
+            end
+            if "author" in keys(data)
+                author = data["author"]
+                push!(notes[i], "by $(author)")
+            end
+            if "lisence" in keys(data)
+                lisence = data["lisence"]
+                push!(notes[i], "$(lisence)")
+            end
+        end
+    end
```

Comment:
Since this is just ad-hoc script, it might be niche picking here, but it might look more concise if we define a function `extract_notes(entry)` and then directly construct and use it in the final for-loop. We actually don't need to introduce this for loop.

Reply:
I got it. I've added `extract_note` and `extract_name` functions.